src/Controller/CompanyController.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace App\Controller;
+
+use App\Entity\Company;
+use App\Form\CompanyType;
+use App\Repository\CompanyRepository;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\IsGranted;
+
+/**
+ * @Route("/admin/company", name="company_")
+ * @IsGranted("ROLE_ADMIN")
+ */
+class CompanyController extends AbstractController
+{
+    /**
+     * @Route("/", name="index", methods={"GET"})
+     */
+    public function index(CompanyRepository $companyRepository): Response
+    {
+        return $this->render('company/index.html.twig', [
+            'companies' => $companyRepository->findAll(),
+        ]);
+    }
+
+    /**
+     * @Route("/new", name="new", methods={"GET","POST"})
+     */
+    public function new(Request $request): Response
+    {
+        $company = new Company();
+        $form = $this->createForm(CompanyType::class, $company);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $entityManager = $this->getDoctrine()->getManager();
+            $entityManager->persist($company);
+            $entityManager->flush();
+
+            return $this->redirectToRoute('company_index');
+        }
+
+        return $this->render('company/new.html.twig', [
+            'company' => $company,
+            'form' => $form->createView(),
+        ]);
+    }
```

Comment:
so an admin can create a new company?

Reply:
Yes, the company ask JobPermut to get an access